manticore/core/manticore.py

```diff
@@ -80,14 +80,18 @@ def __init__(self, initial_state, workspace_url=None, policy='random', **kwargs)
         self._executor.forward_events_from(self._initial_state, True)
 
         # Move the following into a linux plugin
-
         self._assertions = {}
         self._coverage_file = None
         self.trace = None
 
         # FIXME move the following to a plugin
-        self.subscribe('will_generate_testcase', self._generate_testcase_callback)
         self.subscribe('did_finish_run', self._did_finish_run_callback)
+        self.subscribe('internal_generate_testcase', self._publish_generate_testcase)
+
+    def _publish_generate_testcase(self, state, prefix='test', message=''):
```

Comment:
Hmm... we probably don't want child class to be able to override this one. So I am going to change it into `__` instead of `_`.

This _should be obvious_ but honestly, it is easy to forget how some things work.

TLDR: creating an `__attribute` in reality creates a `_CLS_attribute` where `CLS` is the class name and using `self.__attribute` resolves to `_CLS2_attribute` where `CLS2` is the class in which the method that uses the attribute is defined.

An example below.
```
In [10]: class A:
    ...:     def __init__(self):
    ...:         self.x = self._foo
    ...:         self.y = self.__bar
    ...:     def _foo(self):
    ...:         print("A.foo")
    ...:     def __bar(self):
    ...:         print("A.bar")
    ...:
    ...:

In [11]: class B(A):
    ...:     def _foo(self):
    ...:         print("B.foo")
    ...:     def  __bar(self):
    ...:         print("B.bar")
    ...:

In [12]: B().x()
B.foo

In [13]: B().y()
A.bar
```